src/Storages/WindowView/StorageWindowView.cpp

```diff
@@ -447,6 +449,65 @@ bool StorageWindowView::optimize(
     return getInnerStorage()->optimize(query, metadata_snapshot, partition, final, deduplicate, deduplicate_by_columns, local_context);
 }
 
+void StorageWindowView::alter(
+    const AlterCommands & params,
+    ContextPtr local_context,
+    AlterLockHolder &)
+{
+    auto table_id = getStorageID();
+    StorageInMemoryMetadata new_metadata = getInMemoryMetadata();
+    StorageInMemoryMetadata old_metadata = getInMemoryMetadata();
+    params.apply(new_metadata, local_context);
+
+    const auto & new_select = new_metadata.select;
+    const auto & new_select_query = new_metadata.select.inner_query;
+
+    auto old_inner_table_id = inner_table_id;
+
+    modifying_query = true;
```

Comment:
may be better like this?
```suggestion
    modifying_query = true;
    SCOPE_EXIT({
        modifying_query = false;
    });
```